docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1260 @@
+previously saved user data.
+
+Then, the `startUI()` method is called to display the welcome message to the user.
+
+The `CliUserInterface` class then continuously gets the input from the user. It then uses the `InputParser` class to
+parse this input and creates a new `CliCommand` object based on the command entered by the user. It executes the
+command with `CliCommand.execute()` (learn more [here](#33-commands-component-neil)), which will use or modify the
+`AppState` instance to make the requested change.
+
+If any of these steps encounters an error, an exception of the type `ZeroNoteException` is thrown by the method, and
+caught in the `run()` method. Upon catching an exception, the `printErrorMessage()` method is called to display the
+appropriate error message to the user. See [here]() for more information on how exceptions work in **Zer0Note**.
+
+### 3.3. Commands Component (Neil)
+
+Every command that a user can input into **Zer0Note** is represented by an object that extends the abstract class
+`CliCommand`. `CliCommand` contains some basic fields and methods that are shared by all types of commands, such as:
+
+* `execute()`, which is called after setting up the object appropriately, to perform the action requested by the user;
+
+* `appState`, which refers to the `AppState` object that stores the state of the currently running instance of
+**Zer0Note**; and
+
+* `commandParams`, which is a String containing the parameters the user entered as parameters to the command.
+
+The following 2 UML diagrams show the different types of `CliCommand` components used in the application.
+
+This diagram describes the `CliCommand`s related to the Timetable mode.
+
+![UML diagram for Timetable Commands](diagrams/class/jpeg/timetable_commands.jpg)
+
+And this diagram describes the `CliCommand`s related to the Notebook mode.
+
+![UML diagrams for Notebook Commands](diagrams/class/jpeg/notebook_commands.jpg)
+
+// TODO: add the new commands for "Find", "Tag", "SortByDate".
+
+### 3.4. Tasks Component
+
+![UML Diagram from Task Component](diagrams/class/jpeg/taskComponent.jpg)
+
+Figure []. Structure of Tasks Component
+
+The `Tasks` component,
+
+- stores an array of `Task` objects that represents the deadline of the user.
+- together with the `Notebooks` component form the `AppState` component.   
+
+The `TaskList` class,
+
+- has methods to add a new `Task` object and remove existing one at an index.
+
+### 3.5. Notebooks Component
+
+![UML diagram for Notebooks Component](diagrams/class/jpeg/notebooks_simplified.jpg)
+
+Figure []. Structure of Notebooks Component
+
+The `Notebooks` class,
+
+- stores a `NotebookShelf` object that contains a list of `Notebook` objects.
+- together with the `Tasks` component form the `AppState` component.
+
+The `NotebookShelf` class,
+
+- has methods to add new `Notebook` object and remove existing one.
```

Comment:
One suggestion might be to refer to the method names so that it is easier for the developer. It is hard to see from the diagram.